yt/fields/tests/test_field_access_pytest.py

```diff
@@ -0,0 +1,49 @@
+from collections import defaultdict
+from itertools import product
+
+import pytest
+
+from yt.testing import fake_random_ds
+from yt.utilities.exceptions import YTFieldNotFound
+
+
+def test_unexisting_field_access():
+    ds = fake_random_ds(16, particles=10)
+
+    fname2ftype = defaultdict(list)
+
+    for ft, fn in ds.derived_field_list:
+        fname2ftype[fn].append(ft)
+
+    ad = ds.all_data()
+
+    ftypes = ("gas", "io")
+    fnames = (
+        "density",
+        "particle_position_x",
+        "particle_position_y",
+        "particle_position_z",
+    )
+
+    # Try invalid ftypes, fnames combinations
+    for ft, fn in product(ftypes, fnames):
+        if (ft, fn) in ds.derived_field_list:
+            continue
+
+        with pytest.raises(YTFieldNotFound, match="") as excinfo:
```

Comment:
why are you matching an empty string ? it doesn't look like a desirable behaviour to me, why shouldn't the error message exist ? am I missing something ?

Reply:
That's most definitely a typo, and it should be
```suggestion
        with pytest.raises(YTFieldNotFound) as excinfo:
```